### ISSUE TITLE
'atomic images update' pulls latest image even if it is on the host

### ISSUE THREAD

**Issue by miabbott** [projectatomic/atomic#812]
Using atomic 1.14.1, I pulled the latest `nginx` image from docker.io.  Then proceeded to use `atomic images update` multiple times and observed the same image being pulled each time.  Seems like we should be smarter about this when the latest image exists on the host.

```
# atomic pull docker.io/nginx
Image docker.io/nginx is being pulled to docker ...
Pulling docker.io/library/nginx:latest ...
Copying blob sha256:75a822cd7888e394c49828b951061402d31745f596b1f502758570f2d0ee79e2
 48.50 MB / 48.98 MB [=========================================================]
Copying blob sha256:0aefb9dc4a57d3de6a9cfa2e87e4502dfa8ce3876264bb20783b1610f8e44806
 19.23 MB / 19.24 MB [=========================================================]
Copying blob sha256:046e44ee6057f1264d00b0c54adcff2f2c44d30a29b50dfef928776f7aa45cc8
 0 B / 193 B [-----------------------------------------------------------------]
Copying config sha256:01f818af747d88b4ebca7cdabd0c581e406e0e790be72678d257735fad84a15f
 0 B / 3.62 KB [---------------------------------------------------------------]
Writing manifest to image destination
Storing signatures
 3.62 KB / 3.62 KB [===========================================================]

# atomic images list
   REPOSITORY                                                   TAG      IMAGE ID       CREATED            VIRTUAL SIZE   TYPE       
   docker.io/nginx                                              latest   01f818af747d   2016-12-27 17:56   181.59 MB      docker 

# atomic images update docker.io/nginx
Copying blob sha256:75a822cd7888e394c49828b951061402d31745f596b1f502758570f2d0ee79e2
 48.86 MB / 48.98 MB [=========================================================]
Copying blob sha256:0aefb9dc4a57d3de6a9cfa2e87e4502dfa8ce3876264bb20783b1610f8e44806
 19.05 MB / 19.24 MB [=========================================================]
Copying blob sha256:046e44ee6057f1264d00b0c54adcff2f2c44d30a29b50dfef928776f7aa45cc8
 0 B / 193 B [-----------------------------------------------------------------]
Copying config sha256:01f818af747d88b4ebca7cdabd0c581e406e0e790be72678d257735fad84a15f
 0 B / 3.62 KB [---------------------------------------------------------------]
Writing manifest to image destination
Storing signatures

# atomic images list
   REPOSITORY                                                   TAG      IMAGE ID       CREATED            VIRTUAL SIZE   TYPE         
   docker.io/nginx                                              latest   01f818af747d   2016-12-27 17:56   181.59 MB      docker    

# atomic images update docker.io/nginx
Copying blob sha256:75a822cd7888e394c49828b951061402d31745f596b1f502758570f2d0ee79e2
 48.19 MB / 48.98 MB [=========================================================]
Copying blob sha256:0aefb9dc4a57d3de6a9cfa2e87e4502dfa8ce3876264bb20783b1610f8e44806
 19.14 MB / 19.24 MB [=========================================================]
Copying blob sha256:046e44ee6057f1264d00b0c54adcff2f2c44d30a29b50dfef928776f7aa45cc8
 0 B / 193 B [-----------------------------------------------------------------]
Copying config sha256:01f818af747d88b4ebca7cdabd0c581e406e0e790be72678d257735fad84a15f
 0 B / 3.62 KB [---------------------------------------------------------------]
Writing manifest to image destination
Storing signatures

# atomic images list -a
   REPOSITORY                                                   TAG      IMAGE ID       CREATED            VIRTUAL SIZE   TYPE      
   docker.io/nginx                                              latest   01f818af747d   2016-12-27 17:56   181.59 MB      docker
```

Incidentally, this only seems to happen when using docker storage; doing the same with ostree storage is more sane.

```
# atomic pull --storage ostree docker.io/nginx
Image docker.io/nginx is being pulled to ostree ...
Pulling layer 75a822cd7888e394c49828b951061402d31745f596b1f502758570f2d0ee79e2
Pulling layer 0aefb9dc4a57d3de6a9cfa2e87e4502dfa8ce3876264bb20783b1610f8e44806
Pulling layer 046e44ee6057f1264d00b0c54adcff2f2c44d30a29b50dfef928776f7aa45cc8
# atomic images list
   REPOSITORY                                                   TAG      IMAGE ID       CREATED            VIRTUAL SIZE   TYPE      
   docker.io/nginx                                              latest   01f818af747d   2017-01-05 21:56                  ostree    

# atomic images update --storage ostree docker.io/nginx
# echo $?
0
# atomic images list
   REPOSITORY                                                   TAG      IMAGE ID       CREATED            VIRTUAL SIZE   TYPE
   docker.io/nginx                                              latest   01f818af747d   2017-01-05 21:56                  ostree

**Comment by miabbott** [comment]
Heh, looks like @baude may have had the same idea over in projectatomic/skopeo#281

**Comment by baude** [comment]
@miabbott yeah but I'll use this issue to track a fix.  Thanks for reporting.

**Comment by baude** [comment]
should be fixed with https://github.com/projectatomic/atomic/pull/825

**Comment by miabbott** [comment]
Confirmed fixed in #825
